rkyv/src/util/aligned_vec.rs

```diff
@@ -48,13 +48,25 @@ use std::{alloc, io};
 /// assert_eq!(archived.bytes.as_slice(), &[1, 2, 3]);
 /// assert_eq!(archived.bytes.as_ptr().align_offset(16), 0);
 /// ```
-pub struct AlignedVec {
+pub type AlignedVec = AlignedVecCustom<16>;
+
+/// A vector of bytes that aligns its memory to specified alignment.
+///
+/// See `AlignedVec`.
+///
+/// ```
+/// use rkyv::{AlignedVec, AlignedVecCustom};
+///
+/// let bytes = AlignedVecCustom::<4096>::with_capacity(1);
+/// assert_eq!(bytes.as_ptr() as usize % 4096, 0);
+/// ```
+pub struct AlignedVecCustom<const ALIGNMENT: usize> {
```

Comment:
I'd prefer `AlignedByteVec` with a default `const ALIGNMENT: usize = 16` to `AlignedVecCustom`. We could even `#[deprecate]` `AlignedVec`, which I'd be okay with doing.

Reply:
I've made these changes. However, default const parameters aren't very ergonomic without a type alias - you need some type annotation even despite the default:

```rust
// These work
let v = AlignedByteVec::<16>::new();
let v: AlignedByteVec = AlignedByteVec::new();

// Compilation error: type annotations needed for `AlignedByteVec<ALIGNMENT>`
let v = AlignedByteVec::new();
```

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8a161830cd43eafdb8f4860008288a08)

So I think `AlignedVec` needs to stay as a type alias, and personally I don't think there's any need to deprecate it. From my experience, once you go beyond `rkyv::to_bytes` and get into composing a custom serializer, it's quite challenging to figure out all the types. I'm a Rust newbie, so maybe it's just me, but I feel it's better to avoid another compulsory parameter the user has to figure out before their code will compile.

What do you think?

Reply:
Ah I see the issue with default const parameters. I'm fine with leaving it un-deprecated, I'll probably remove the type alias with the next minor version bump when I can do semver-breaking changes.